src/easylink/configuration.py

```diff
@@ -173,6 +173,14 @@ def update_implementation_configs(self, level: LayeredConfigTree):
 
             elif "substeps" in step:
                 self.update_implementation_configs(step.substeps)
+            elif "iterate" in step:
+                for idx, loop in enumerate(step.iterate):
+                    loop = LayeredConfigTree(loop)
+                    step.iterate[idx].update(loop)
+                    if "implementation" in loop:
+                        loop.implementation.update({"configuration": {}}, layer="default")
+                    else:
+                        self.update_implementation_configs(loop)
```

Comment:
Just an idea, but could this method be simplified by making `{}` the default configuration at the time when it is accessed, rather than upfront on the LayeredConfigTree? It feels like it is starting to get complicated.

Reply:
agreed

Reply:
https://github.com/ihmeuw/easylink/pull/99/files